File: tools/update_words.js
const fs = require('fs');
const path = require('path');

// 配置
const DATA_DIR = path.join(__dirname, '../data');
const UNSPLASH_BASE_URL = 'https://images.unsplash.com/photo-';
const UNSPLASH_SIZE = 'w=400';

// 读取JSON文件
function readJsonFile(filePath) {
    try {
        const content = fs.readFileSync(filePath, 'utf8');
        return JSON.parse(content);
    } catch (error) {
        console.error(`Error reading file ${filePath}:`, error.message);
        return null;
    }
}

// 写入JSON文件
function writeJsonFile(filePath, data) {
    try {
        const content = JSON.stringify(data, null, 4);
        fs.writeFileSync(filePath, content, 'utf8');
        console.log(`Successfully wrote to ${filePath}`);
    } catch (error) {
        console.error(`Error writing file ${filePath}:`, error.message);
    }
}

// 生成Unsplash图片URL
function generateImageUrl() {
    const randomId = Math.floor(Math.random() * 1000000000);
    return `${UNSPLASH_BASE_URL}${randomId}?${UNSPLASH_SIZE}`;
}

// 更新单词文件
function updateWordFiles() {
    // 读取所有分类信息
    const categories = readJsonFile(path.join(DATA_DIR, 'all_categories.json'));
    if (!categories) return;

    // 读取主单词文件
    const mainWords = readJsonFile(path.join(DATA_DIR, 'wordlist.json'));
    if (!mainWords) return;

    // 为每个单词添加分类和图片
    const updatedMainWords = mainWords.map(word => {
        // 如果单词没有图片，添加一个
        if (!word.image) {
            word.image = generateImageUrl();
        }
        return word;
    });

    // 更新主文件
    writeJsonFile(path.join(DATA_DIR, 'wordlist.json'), updatedMainWords);

    // 更新每个分类文件
    categories.forEach(category => {
        const categoryFile = path.join(DATA_DIR, category.file);
        const categoryWords = readJsonFile(categoryFile);
        if (categoryWords) {
            // 更新分类文件中的单词
            const updatedCategoryWords = categoryWords.map(word => {
                // 如果单词没有图片，添加一个
                if (!word.image) {
                    word.image = generateImageUrl();
                }
                return word;
            });
            writeJsonFile(categoryFile, updatedCategoryWords);
            
            // 更新分类信息中的单词数量
            category.count = updatedCategoryWords.length;
        }
    });

    // 更新分类信息文件
    writeJsonFile(path.join(DATA_DIR, 'all_categories.json'), categories);
}

// 运行更新
updateWordFiles(); 

File: tools/update_words.py
import json
import os
import random
from pathlib import Path

# 配置
DATA_DIR = Path(__file__).parent.parent / 'data'
UNSPLASH_BASE_URL = 'https://images.unsplash.com/photo-'
UNSPLASH_SIZE = 'w=400'

def read_json_file(file_path):
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            return json.load(f)
    except Exception as e:
        print(f'Error reading file {file_path}: {str(e)}')
        return None

def write_json_file(file_path, data):
    try:
        with open(file_path, 'w', encoding='utf-8') as f:
            json.dump(data, f, ensure_ascii=False, indent=4)
        print(f'Successfully wrote to {file_path}')
    except Exception as e:
        print(f'Error writing file {file_path}: {str(e)}')

def generate_image_url():
    random_id = random.randint(100000000, 999999999)
    return f'{UNSPLASH_BASE_URL}{random_id}?{UNSPLASH_SIZE}'

def update_word_files():
    # 读取所有分类信息
    categories = read_json_file(DATA_DIR / 'all_categories.json')
    if not categories:
        return

    # 读取主单词文件
    main_words = read_json_file(DATA_DIR / 'wordlist.json')
    if not main_words:
        return

    # 为每个单词添加图片
    updated_main_words = []
    for word in main_words:
        if not word.get('image'):
            word['image'] = generate_image_url()
        updated_main_words.append(word)

    # 更新主文件
    write_json_file(DATA_DIR / 'wordlist.json', updated_main_words)

    # 更新每个分类文件
    for category in categories:
        category_file = DATA_DIR / category['file']
        category_words = read_json_file(category_file)
        if category_words:
            # 更新分类文件中的单词
            updated_category_words = []
            for word in category_words:
                if not word.get('image'):
                    word['image'] = generate_image_url()
                updated_category_words.append(word)
            write_json_file(category_file, updated_category_words)
            
            # 更新分类信息中的单词数量
            category['count'] = len(updated_category_words)

    # 更新分类信息文件
    write_json_file(DATA_DIR / 'all_categories.json', categories)

if __name__ == '__main__':
    update_word_files() 